WixEmbedsAPI docs (#16)
wix-incubator/wix-client-docs: apis/guides/WixEmbedsAPI.md

> # WixEmbedsAPI

Reviewer: Maybe this should be "Wix Embeds"?

> The `wixEmbedsApi` is a property of the global `window` javascript object.

Reviewer: In the code examples `API` is (regrettably) in all caps. Which one is right?

> The `wixEmbedsApi` is a property of the global `window` javascript object.

Reviewer: The first line here should describe what this API is. I think you can leave the fact that it's part of `window` for the usage section.

> Use this API in [self-hosted Embedded Script extensions](https://dev.wix.com/docs/build-apps/develop-your-app/frameworks/self-hosting/supported-extensions/site-extensions/add-an-embedded-script-extension-to-a-self-hosted-app) to interact with the site's page.

Reviewer: Can we get any more specific than "interact with the site's pages"?

Author: I'm struggling to think of something more specific given that it has lots of functionality, but right now we're only releasing one function. So either it'll be accurate now, I could write something that will be accurate in the future, but struggling to do both

Reviewer: Ok, so let's leave it for now. But maybe change to "interact with a site's pages".

## Usage

> You don't need to import `wixEmbedsApi` because it's a property of the global `window` javascript object. However, you can't immediately access it when your embedded script runs.

Reviewer: ```suggestion
You don't need to import `wixEmbedsApi` because it's a property of the global JavaScript `window` object. However, you can't immediately access it when your embedded script runs.
```

Therefore, you should add an event listener that listens for the `wixEmbedsAPIReady` event.

For example:
```js
if (window.wixEmbedsAPI) {
  // Run your code that calls a wixEmbedsAPI method here.
} else {
  window.addEventListener(
    "wixEmbedsAPIReady",
    () => {
      // Run your code that calls a wixEmbedsAPI method here.
    },
    false,
  );
}
```

You could also store a boolean variable that sets to `true` on `wixEmbedsAPIReady` and check the variable value before attempting to call a `wixEmbedsAPI` method.

> ## Methods

Reviewer: We're sort of breaking new ground in this reference, and I think it makes sense to stay aligned with what we have in the other refs. So I suggest breaking this out into a separate 'function' article and having the first part as an intro article.

WDYT?

Author: In the Client portal, the API articles have multiple functions in and are more similar to this layout

### getAccessTokenFunction()

> Returns an asynchronous function that returns a promise that resolves to an access token.

Reviewer: ```suggestion
Returns an asynchronous function that resolves to an access token.
```

Reviewer: Every asynchronous function in JS returns a promise.

> Use this access token to call Wix [REST APIs](https://dev.wix.com/docs/rest) and [SDKs](https://dev.wix.com/docs/sdk) in an [embedded script](https://dev.wix.com/docs/build-apps/develop-your-app/frameworks/self-hosting/supported-extensions/site-extensions/add-an-embedded-script-extension-to-a-self-hosted-app) app extension. Learn more about [authorizing API calls in self-hosted embedded script extensions](ADD-LINK-TO-KB-ARTICLE).

Reviewer: ```suggestion
Use this access token to call Wix [REST APIs](https://dev.wix.com/docs/rest) and use the Wix [JavaScript SDK](https://dev.wix.com/docs/sdk) in an [embedded script](https://dev.wix.com/docs/build-apps/develop-your-app/frameworks/self-hosting/supported-extensions/site-extensions/add-an-embedded-script-extension-to-a-self-hosted-app) app extension. 

Learn more about [authorizing API calls in self-hosted embedded script extensions](ADD-LINK-TO-KB-ARTICLE).
```

> 

Reviewer: What kind of access token is this? Which identity does it use? Is that info included in the article you're linking out to?

Author: As this function is for use in self hosted apps, does that not mean it's site visitor auth?
In the [REST access token docs](https://dev.wix.com/docs/rest/app-management/oauth-2/introduction), it doesn't specify either the identity. Should we still include here?

Reviewer: My guess is that it would be either site visitor or site member. Maybe we should mention that the access token allows you to authenticate as a site visitor or site member and link to [here](https://dev.wix.com/docs/build-apps/develop-your-app/access/about-identities)?

#### Syntax
```js
getAccessTokenFunction(): () => Promise<accessToken>
```

#### Returns

> Asynchronous function that returns a promise that resolves to an access token.

Reviewer: ```suggestion
Asynchronous function that resolves to an access token.
```

> #### Code sample

Reviewer: ```suggestion
#### Examples
```

> ```js

Reviewer: Include a header with a title for the example (e.g., "Retrieve an access token").
